testsuite/smoke-test/src/client.rs

```diff
@@ -163,62 +168,73 @@ fn test_basic_fault_tolerance() {
 #[test]
 fn test_basic_restartability() {
     let mut swarm = new_local_swarm(4);
-    let client = swarm.validators().next().unwrap().json_rpc_client();
+    let client = swarm.validators().next().unwrap().rest_client();
     let transaction_factory = swarm.chain_info().transaction_factory();
 
     let mut account_0 = create_and_fund_account(&mut swarm, 100);
     let account_1 = create_and_fund_account(&mut swarm, 10);
-    transfer_coins(
-        &client,
-        &transaction_factory,
-        &mut account_0,
-        &account_1,
-        10,
-    );
-    assert_balance(&client, &account_0, 90);
-    assert_balance(&client, &account_1, 20);
+
+    let runtime = Runtime::new().unwrap();
```

Comment:
This change would be so much cleaner/simpler if we didn't have to create a new Runtime in every smoke test and use that to call `block_on(...)`. Is there no better way? Some ideas might be:
- Use `[tokio::test]` instead of `[test]` to allow tokio to create the runtime? Would that work or am I missing something?
- Create a runtime in the helpers (e.g., `transfer_coins` and `assert_balance`) and block on those instead of doing it at the call sites?

Reply:
we need to do this step by step, I'm converting util functions into async first, once we converted all functions into async, then we can change [test] to [tokio::test].
The reason we need to do this, is because when you have a tokio runtime wraps an async call already to make a function sync, you can't wrap the function into another tokio runtime anymore.
Technically I can try out just convert tests into [tokio::test], and find out what can and what can't, but it requires a lot more time to do so, because once I start to convert everything into async and found something not work because of a wrapped tokio runtime I need convert them back.

To avoid that, I just go for migrating small pieces first, then when we found runtime doing all of block_on in the test, we can then convert test into [tokio::test] easily.

Reply:
Yeah Xiao and I already talked about this and how we'll need to iteratively improve all of these things. But this is a good first step

Reply:
Cool, thanks folks! I guess my strategy would have been to do things the other way around (i.e., change `[test]` to `[tokio::test]` and then see what broke, but as long as we'll come back to clean this stuff up, I'm happy!